Reflect.js

```diff
@@ -813,7 +813,7 @@ var Reflect;
         // https://tc39.github.io/ecma262/#sec-isconstructor
         function IsConstructor(argument) {
             // NOTE: This is an approximation as we cannot check for [[Construct]] internal method.
-            return typeof argument === "function";
+            return typeof argument === "function" || argument.constructor;
```

Comment:
This is incorrect. *every* object (except for an object with a `null` \[\[Prototype]] slot) has a `constructor` property.

Reply:
But the problem is that not all decorators returns a function (hence typeof argument) is not always a function. For instance, if you use connect from react-redux, you will see that the target is not a function but an object.

Reply:
All "class decorators" should accept a `function` and return either a `function` (for constructor replacement), or `undefined`/`void` (when no constructor replacement should occur).

All "method decorators" should accept an `object` (or `function`), a property key (`string` or `symbol`), and a `PropertyDescriptor`, and return either a `PropertyDescriptor` (for member replacment), or `undefined`/`void` (when no member replacement should occur). 

All "field decorators" should accept an `object` (or `function`) and a property key (`string` or `symbol`), and should return `undefined`/`void`.

The `IsConstructor` function is used to test whether a "class decorator" has an appropriate target. If you are running into an issue, it could be a result of how the metadata functions are being called. I would need to see a repro of the issue to suggest an appropriate fix.

Reply:
You are right. And the issue is not with the IsConstructor, but where it is used (my bad). Please review the following issue: https://github.com/rbuckton/reflect-metadata/issues/108